yabt/builders/golang.py

```diff
@@ -191,43 +213,53 @@ def go_builder_internal(build_context, target, command):
     # GOPATH but if it come handy we should implement looking into the docker
     gopaths.append(user_gopath if user_gopath else '/go')
     build_cmd_env = {
-        'XDG_CACHE_HOME': '/tmp/.cache',
+        'XDG_CACHE_HOME': download_cache_dir,
     }
     build_cmd_env.update(target.props.cmd_env or {})
     build_cmd_env['GOPATH'] = ':'.join(gopaths)
 
+    go_mod_path = join(workspace_dir, 'go.mod')
+    user_mod_path = target.props.get('mod_file', None) or build_context.conf.get('go_mod_file', None)
```

Comment:
I thought we are not supporting mod_file per target, no?

Reply:
If so, does it have meaning in a GoPackage target? how does it behave?

Reply:
Since the GoPackage target builds its own module tree without a main, it behaves the same. This is to give the option for target to have its own version dependency, but I guess this has no effect, since the library files are linked to any target depending on them and in this case use this target mod file and not the library. Can have an effect on binaries that can depend on different versions of external modules.

Reply:
I think that if we allow GoPackage to go.mod for pinning a version or whatever other things that can be done with go.mod, then we have to do some sort of merging of go.mod files (and exit with error on unmergeable conflicts).

My vote is to remove the mod_file from GoPackage signature

Reply:
Done.